Missing header (`config.hpp`) from the boost package

<!-- Thank you for contributing to osquery! -->

# Bug report
Tried building osquery-extension which uses boost process library on macOS. It fails due to missing file `boost/winapi/config.hpp`. The file is not included on the non-windows platform. 

```
In file included from .../osquery_workspace/osquery/libraries/cmake/source/boost/src/libs/process/include/boost/process.hpp:27:
.../osquery_workspace/osquery/libraries/cmake/source/boost/src/libs/process/include/boost/process/cmd.hpp:14:10: fatal error: 
   'boost/winapi/config.hpp' file not found
#include <boost/winapi/config.hpp>
     ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
### What operating system and version are you using?
MacOS 10.15

### What version of osquery are you using?
master

--- a/libraries/cmake/source/boost/CMakeLists.txt
+++ b/libraries/cmake/source/boost/CMakeLists.txt
@@ -275,20 +275,18 @@ endfunction()
 function(generateBoostWinapi)
   add_library(thirdparty_boost_winapi INTERFACE)
 
-  if(NOT DEFINED PLATFORM_WINDOWS)
-    return()
-  endif()
-
   set(library_root "${BOOST_ROOT}/libs/winapi")
 
   target_include_directories(thirdparty_boost_winapi SYSTEM INTERFACE
     "${library_root}/include"
   )
 
-  target_compile_definitions(thirdparty_boost_winapi INTERFACE
-    BOOST_USE_WINDOWS_H
-    NOMINMAX
-  )
+  if(DEFINED PLATFORM_WINDOWS)
+    target_compile_definitions(thirdparty_boost_winapi INTERFACE
+      BOOST_USE_WINDOWS_H
+      NOMINMAX
+    )
+  endif()
 
   target_link_libraries(thirdparty_boost_winapi INTERFACE
     thirdparty_boost_config

